api/src/Validator/AllowTransition/AssertAllowTransitionsValidator.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace App\Validator\AllowTransition;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\InvalidArgumentException;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+class AssertAllowTransitionsValidator extends ConstraintValidator {
+    public function __construct(public RequestStack $requestStack) {
+    }
+
+    public function validate($value, Constraint $constraint) {
+        if (!$constraint instanceof AssertAllowTransitions) {
+            throw new UnexpectedTypeException($constraint, AssertAllowTransitions::class);
+        }
+
+        $transitions = $constraint->getTransitions();
+        if (!$transitions->forAll(fn ($_, $elem) => self::isValidTransition($elem))) {
+            throw new InvalidArgumentException('All transitions must have a from and a to key');
+        }
+
+        $transitions = $transitions->map(fn ($elem) => new Transition($elem['from'], $elem['to']));
+
+        $allFrom = $transitions->map(fn (Transition $elem) => $elem->from)->toArray();
+        $allTo = $transitions->map(fn (Transition $elem) => $elem->to)->toArray();
```

Comment:
Brauchts diesen Zwischenschritt und die `Transition` Klasse wirklich? Warum nicht gleich direkt so:
```suggestion
        $allFrom = $transitions->map(fn ($elem) => $elem['from'])->toArray();
        $allTo = $transitions->map(fn ($elem) => $elem['from'])->toArray();
```
(und weiter unten auch noch `$transition->from` und `$transition->to` anpassen)

Reply:
done